### PR TITLE
[GITHUB-14] Update to Latest Standards

### PR DESCRIPTION
**Pull request by dennisconrad**
* Flatten variables
* Convert to molecule testing
* Solves #6 
* Supersedes #10

### CODE REVIEW

File: tasks/build.yml [hunk 1]
```diff
@@ -1,28 +1,37 @@
...
-  args:
-    chdir: /tmp
-    creates: /tmp/nodesource.gpg.key
+- name: Install Python Modules
+  become: yes
+  pip:
+    name: "{{ item }}"
+    state: present
+  with_items:
+    - pip==9.0.3
```
Comment (lobsterdore) [inline tasks/build.yml]: I feel like the versions should be configurable, why have these modules been added anywho? To get around SSL errors?

Comment (dennisconrad) [reply tasks/build.yml]: `pip` and `setuptools` of at least those versions are required to install `ndg-httpsclient`, which in turn is required to validate https://deb.nodesource.com/

Comment (dennisconrad) [reply tasks/build.yml]: I have change it to "greater than", which should be better and not overwrite modules already installed, as long as they have a minimum version.  As this role has nothing directly to do with python, and only requires this to validate the nodejs repo, I feel python modules shouldn't be configurable in the role.

**Comment by lobsterdore** [comment]
LGTM 👍

**Review by rsgoodall — APPROVED** [review]

**Pull request merged**